fili-core/src/test/groovy/com/yahoo/bard/webservice/data/SerializationResources.groovy

```diff
@@ -144,35 +143,36 @@ class SerializationResources extends Specification {
         schema3 = buildSchema(baseSchemaTypeMap)
 
         List<Result> results = new ArrayList<>([result1, result2])
-        resultSet = new ResultSet(results, schema)
+        resultSet = new ResultSet(schema, results)
 
-        DateTime ny = new DateTime(2011, 2, 2, 7, 0, 0, 0, DateTimeZone.forID("UTC"));
-        DateTime la = new DateTime(2011, 2, 3, 10, 15, 0, 0, DateTimeZone.forID("UTC"));
+        DateTime ny = new DateTime(2011, 2, 2, 7, 0, 0, 0, DateTimeZone.forID("UTC"))
+        DateTime la = new DateTime(2011, 2, 3, 10, 15, 0, 0, DateTimeZone.forID("UTC"))
         interval = new Interval(ny, la)
         bigDecimal = new BigDecimal("100")
 
         responseContext = new ResponseContext([:])
         responseContext.put("randomHeader", "someHeader")
-        responseContext.put("missingIntervals", ["a","b","c", new SimplifiedIntervalList([interval]), bigDecimal])
+        responseContext.put("missingIntervals", ["a","b","c", new SimplifiedIntervalList([interval]), bigDecimal] as ArrayList)
 
         responseContext1 = new ResponseContext([:])
         responseContext1.put("randomHeader", "someHeader")
-        responseContext1.put("apiMetricColumnNames", ["metric1, metric2"] as Set)
+        responseContext1.put("apiMetricColumnNames", ["metric1, metric2"] as LinkedHashSet)
         responseContext1.put("requestedApiDimensionFields", [(ageBracketDim.getApiName()) : [BardDimensionField.ID] as Set])
 
         preResponse = new PreResponse(resultSet, responseContext)
 
         return this
     }
 
-    ZonedSchema buildSchema(Map<String, String> metricNameClassNames) {
-        Schema schema = new ZonedSchema(granularity, DateTimeZone.UTC)
-        schema.addColumn(new DimensionColumn(dimensionDictionary.findByApiName("ageBracket")))
-        schema.addColumn(new DimensionColumn(dimensionDictionary.findByApiName("gender")))
-        schema.addColumn(new DimensionColumn(dimensionDictionary.findByApiName("country")))
+    ResultSetSchema buildSchema(Map<String, String> metricNameClassNames) {
+        LinkedHashSet columns = []
+        columns << new DimensionColumn(dimensionDictionary.findByApiName("ageBracket"))
+        columns << new DimensionColumn(dimensionDictionary.findByApiName("gender"))
+        columns << new DimensionColumn(dimensionDictionary.findByApiName("country"))
         metricNameClassNames.each {
-             schema.addColumn(new MetricColumnWithValueType(it.key, it.value))
+             columns << new MetricColumnWithValueType(it.key, it.value)
```

Comment:
Why are we using `<<` instead of `add` here? It just feels strange suddenly using the `<<` operator which I believe is rarely found in our code?